### PR TITLE
Abdul Majid

### PR DESCRIPTION
**Pull request by ghost**
Please review

### CODE REVIEW

File: src/main/java/com/tradingengine/marketdataservice/integrations/ExchangeUrls.java
Comment (kwame-one) [inline src/main/java/com/tradingengine/marketdataservice/integrations/ExchangeUrls.java:1]: I don't think you need all the endpoints here. The base URLs should be fine

File: src/main/java/com/tradingengine/marketdataservice/integrations/FetchExchangeDataFromAPIIntegration.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    private final ObjectMapper mapper;
+
+    public FetchExchangeDataFromAPIIntegration(RestTemplate restTemplate, ObjectMapper mapper) {
+        this.restTemplate = restTemplate;
+        this.mapper = mapper;
+    }
+
+    public List<OrderDto> getOpenOrders(String exchangeUrl) {
+        return   restTemplate.exchange(exchangeUrl, HttpMethod.GET, null,
+                new ParameterizedTypeReference<List<OrderDto>>() {}).getBody();
```
Comment (kwame-one) [inline src/main/java/com/tradingengine/marketdataservice/integrations/FetchExchangeDataFromAPIIntegration.java]: This looks good but always think about edge cases too. I don't see any exception-handling mechanism here.
If you're intercepting it someone in the app, then that's fine

File: src/main/java/com/tradingengine/marketdataservice/integrations/FetchExchangeDataFromAPIIntegration.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    public List<MarketDataDto> getMarketData(String exchangeUrl) {
+        return restTemplate.exchange(exchangeUrl, HttpMethod.GET, null,
```
Comment (kwame-one) [inline src/main/java/com/tradingengine/marketdataservice/integrations/FetchExchangeDataFromAPIIntegration.java]: no exception handling mechanism

**Review by Rose-AM — APPROVED** [review]
nice work

**Pull request merged**